timechop/timechop.py

```diff
@@ -25,21 +26,29 @@ def chop_time(self):
                 )
         return(matrix_set_definitions)
 
-    def calculate_matrix_end_times(self):
+    def calculate_matrix_end_times(self, look_back_duration):
         matrix_end_times = []
         update_delta = utils.convert_str_to_relativedelta(self.update_window)
         matrix_end_time = self.modeling_end_time - update_delta
-        while matrix_end_time > self.modeling_start_time:
+        look_back_delta = utils.convert_str_to_relativedelta(look_back_duration)
+        matrix_start_time = matrix_end_time - look_back_delta
+        if matrix_start_time < self.modeling_start_time:
+            raise ValueError('''
+                Start of first matrix, {}, is before modeling start time, {}.
+            '''.format(matrix_start_time, self.modeling_start_time))
+
+        while matrix_start_time >= self.modeling_start_time:
             matrix_end_times.insert(0, matrix_end_time)
             matrix_end_time -= update_delta
+            matrix_start_time = matrix_end_time - look_back_delta
 
-        if matrix_end_time == self.modeling_start_time:
-            return(matrix_end_times)
-        else:
-            raise ValueError('''Modeling period not evenly divisbile by update
+        if matrix_end_time != self.modeling_start_time:
+            warnings.warn('''Modeling period not evenly divisbile by update
```

Comment:
Oh, thanks! I learned a lot from that. I think I agree with your conclusion here. :)